code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm

```diff
@@ -123,59 +75,78 @@ Difficulty: Hard
+	chosen_attack_num = 2
+
+/datum/action/innate/megafauna_attack/hallucination_surround
+	name = "Surround Target"
+	icon_icon = 'icons/turf/walls/wall.dmi'
+	button_icon_state = "wall"
+	chosen_message = "<span class='colossus'>You are now surrounding the target you click on with hallucinations.</span>"
+	chosen_attack_num = 3
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/do_attack_animation(atom/A, visual_effect_icon)
-	if(!charging)
-		..()
-
-/mob/living/simple_animal/hostile/megafauna/bubblegum/AttackingTarget()
-	if(!charging)
-		. = ..()
-		if(.)
-			recovery_time = world.time + 20 // can only attack melee once every 2 seconds but rapid_melee gives higher priority
+/mob/living/simple_animal/hostile/megafauna/bubblegum/OpenFire()
+	if(charging)
+		return
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/bullet_act(obj/item/projectile/P)
-	if(is_enraged())
-		visible_message("<span class='danger'>[src] deflects the projectile; [p_they()] can't be hit with ranged weapons while enraged!</span>", "<span class='userdanger'>You deflect the projectile!</span>")
-		playsound(src, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 300, 1)
-		return 0
-	return ..()
+	anger_modifier = CLAMP(((maxHealth - health)/60),0,20)
+	ranged_cooldown = world.time + 50
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/ex_act(severity, target)
-	if(severity >= EXPLODE_LIGHT)
+	if(client)
+		if(chosen_attack == 1)
+			triple_charge()
+		else if(chosen_attack == 2)
+			hallucination_charge()
+		else if(chosen_attack == 3)
+			surround_with_hallucinations()
 		return
-	severity = EXPLODE_LIGHT // puny mortals
-	return ..()
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/CanPass(atom/movable/mover, turf/target)
-	if(istype(mover, /mob/living/simple_animal/hostile/megafauna/bubblegum/hallucination))
-		return 1
-	return ..()
+	if(!try_bloodattack())
+		blood_warp()
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/Goto(target, delay, minimum_distance)
-	if(!charging)
-		..()
+	if(health > maxHealth * 0.5)
+		if(prob(50 + anger_modifier))
+			triple_charge()
+		else
+			hallucination_charge()
+	else
+		if(prob(50 - anger_modifier))
+			hallucination_charge()
+		else
+			surround_with_hallucinations()
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/MoveToTarget(list/possible_targets)
-	if(!charging)
-		..()
+/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/triple_charge()
+	charge(delay = 6)
+	charge(delay = 4) // The FitnessGram Pacer Test is a multistage aerobic capacity test that progressively gets more difficult as it continues.
+	charge(delay = 2)
+	SetRecoveryTime(15)
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/Move()
-	if(charging)
-		new /obj/effect/temp_visual/decoy/fading(loc,src)
-		DestroySurroundings()
-	..()
+/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/hallucination_charge()
+	if(health > maxHealth * 0.5)
```

Comment:
if bubblegum's mechanic is he's empowered after an HP threshold it would be nice if that was defined so if I wanted to change it to say just 30% of his HP I could easily do so

`#define BUBBLEGUM_SMASH (health > maxhealth*0.5)` or something

Reply:
if each attack has a different threshold that just happens to be the same value in some cases then this can be ignored ofc